cmd/dep.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2017 The kubecfg authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ksonnet/ksonnet/metadata"
+	"github.com/spf13/cobra"
+)
+
+const (
+	flagName = "name"
+)
+
+func init() {
+	RootCmd.AddCommand(depCmd)
+	depCmd.AddCommand(depAddCmd)
+	depCmd.AddCommand(depListCmd)
+	depAddCmd.PersistentFlags().String(flagName, "", "Name to give the dependency")
+}
+
+var depCmd = &cobra.Command{
+	Use:   "dep",
+	Short: `Manage dependencies for the current ksonnet project`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return fmt.Errorf("Command 'dep' requires a subcommand\n\n%s", cmd.UsageString())
+	},
+	Long: ``,
+
+	Example: ``,
+}
+
+var depAddCmd = &cobra.Command{
+	Use:   "add <registry>/<library>@<version>",
+	Short: `Add a dependency to current ksonnet application`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return fmt.Errorf("Command 'dep add' requires a single argument of the form <registry>/<library>@<version>")
+		}
+
+		registry, libID, name, version, err := parseDepSpec(cmd, args[0])
+		if err != nil {
+			return err
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		wd := metadata.AbsPath(cwd)
+
+		manager, err := metadata.Find(wd)
+		if err != nil {
+			return err
+		}
+
+		_, err = manager.CacheDependency(registry, libID, name, version)
```

Comment:
Did you want to move the non-command parsing logic over to the `/kubecfg/` package? Same applies for the other commands.

Reply:
Let's defer to after presentation.

Reply:
https://github.com/ksonnet/ksonnet/issues/65